src/components/Calculator.js

```diff
@@ -3,88 +3,80 @@ import React from 'react';
-        <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { this.calculate(calculatorObj, '5'); }} onKeyPress={() => { this.calculate(calculatorObj, '5'); }}>5</div>
-        <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { this.calculate(calculatorObj, '6'); }} onKeyPress={() => { this.calculate(calculatorObj, '6'); }}>6</div>
-        <div className="operators-buttons calc-button-style" role="button" tabIndex="0" onClick={() => { this.calculate(calculatorObj, '-'); }} onKeyPress={() => { this.calculate(calculatorObj, '-'); }}>-</div>
-        <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { this.calculate(calculatorObj, '1'); }} onKeyPress={() => { this.calculate(calculatorObj, '1'); }}>1</div>
-        <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { this.calculate(calculatorObj, '2'); }} onKeyPress={() => { this.calculate(calculatorObj, '2'); }}>2</div>
-        <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { this.calculate(calculatorObj, '3'); }} onKeyPress={() => { this.calculate(calculatorObj, '3'); }}>3</div>
-        <div className="operators-buttons calc-button-style" role="button" tabIndex="0" onClick={() => { this.calculate(calculatorObj, '+'); }} onKeyPress={() => { this.calculate(calculatorObj, '+'); }}>+</div>
-        <div className="zero-button calc-button-style" role="button" tabIndex="0" onClick={() => { this.calculate(calculatorObj, '0'); }} onKeyPress={() => { this.calculate(calculatorObj, '0'); }}>0</div>
-        <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { this.calculate(calculatorObj, '.'); }} onKeyPress={() => { this.calculate(calculatorObj, '.'); }}>.</div>
-        <div className="operators-buttons calc-button-style" role="button" tabIndex="0" onClick={() => { this.calculate(calculatorObj, '='); }} onKeyPress={() => { this.calculate(calculatorObj, '='); }}>=</div>
+  return (
+    <div id="calculator-container" className="calculator-container">
+      <div id="result" className="result-display calc-button-style">
+        { displayResult(calculatorObj) }
       </div>
-    );
-  }
+      <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, 'AC', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, 'AC', setCalculatorObj); }}>AC</div>
+      <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, '+/-', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, '+/-', setCalculatorObj); }}>+/-</div>
+      <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, '%', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, '%', setCalculatorObj); }}>%</div>
```

Comment:
- [ ] [OPTIONAL] I think you can avoid the repetition in this component using arrays and the map function, this can keep the code clean, you can have something like this: 
```
const rawOne = ['AC', '+/-', '%', '÷']

// in the render method
{rawOne.map(item => (
       <div className="calc-button-style" role="button" tabIndex="0" 
          onClick={() => { updateCalculation(calculatorObj, item , setCalculatorObj); }} 
          onKeyPress={() => { updateCalculation(calculatorObj, item, setCalculatorObj); }}>
              {item}
       </div> 
))}

```
Then you can do the same for the other raws and that will keep the code clean 😉

Reply:
Thanks for the direction. I thought about it, but I need to understand it a little better to further reduce my code. I will try this ASAP this.